### PR TITLE
Feat job seeker UI

### PR DESCRIPTION
**Pull request by devsharmagit**
closes the issue #548 

https://github.com/user-attachments/assets/72f4dcf0-a4f4-4f6f-9ced-9ad3d5a3da15

### CODE REVIEW

**Comment by devsharmagit** [comment]
@VineeTagarwaL-code let me know if it requires more changes or not

**Comment by VineeTagarwaL-code** [comment]
Cool pr, had a good chat in discord dms ... waiting for the small addition asked then will review the whole code

**Comment by VineeTagarwaL-code** [comment]
this will be little time taking and constant request response will be needed, so please bear with me for the time being 

Thanks again !

**Comment by devsharmagit** [comment]
@VineeTagarwaL-code 
okay sir
give me any changes you want, I will update them as soon as i can.

File: src/components/profile/AccountDeleteDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react';
+import { deleteExperience } from '@/actions/user.profile.actions';
```
Comment (VineeTagarwaL-code) [inline src/components/profile/AccountDeleteDialog.tsx:2]: this is experience delete dialog box and the name of the file is accountdeletedialog


misleading ig

File: src/components/profile/AccountDeleteDialog.tsx
Comment (VineeTagarwaL-code) [inline src/components/profile/AccountDeleteDialog.tsx:1]: you have a file called expdeleteDialog in which you call deleteExp()

and same you are doing here, are you sure this is correct ?

File: src/components/profile/ProfileAboutMe.tsx
Comment (VineeTagarwaL-code) [inline src/components/profile/ProfileAboutMe.tsx:1]: the naming of this file is not proper it should aboutMe only

**Review by VineeTagarwaL-code — CHANGES_REQUESTED** [review]

**Comment by devsharmagit** [comment]
@VineeTagarwaL-code 
Okay sir 
Will update them as soon as possible

File: src/actions/auth.actions.ts [hunk 2]
```diff
@@ -42,7 +42,8 @@ export const signUp = withServerActionAsyncCatcher<
     await prisma.$transaction(
       async (txn) => {
         const user = await txn.user.create({
-          data: { ...data, password: hashedPassword },
+          // todo username
+          data: { ...data, password: hashedPassword, username: 'asdif' },
```
[the remark this answers is not in the document]
Comment (devsharmagit) [reply src/actions/auth.actions.ts:46]: username was present in figma. changing schema leads to require username whenever user is being created. we can remove username from schema or need to add username input in signup form.

File: prisma/schema.prisma [hunk 3]
```diff
@@ -120,6 +147,7 @@ model Project {
   stack            ProjectStack @default(OTHERS)
   userId           String
   user             User         @relation(fields: [userId], references: [id], onDelete: Cascade)
+  isFeature        Boolean @default(false)
```
[the remark this answers is not in the document]
Comment (devsharmagit) [reply prisma/schema.prisma:150]: image

there was a "Feature this project" in the figma, so i added this field in the schema. If the isFeature is set to true then the project will show up other it will hide under show more button

File: src/components/profile/AccountDeleteDialog.tsx (hunk 1 above)
Comment (devsharmagit) [reply src/components/profile/AccountDeleteDialog.tsx:2]: sorry this file is mistakenly committed 
this component is not used anywhere.

File: src/components/profile/AccountDeleteDialog.tsx
Comment (devsharmagit) [reply src/components/profile/AccountDeleteDialog.tsx:1]: sorry this file is mistakenly committed
this component is not used anywhere.

File: src/components/profile/ProfileAboutMe.tsx
Comment (devsharmagit) [reply src/components/profile/ProfileAboutMe.tsx:1]: okay changing the name of the file

**Comment by devsharmagit** [comment]
@VineeTagarwaL-code 
now social links are optional

https://github.com/user-attachments/assets/2830f31c-b476-4579-9b41-a890a1eec431

**Comment by VineeTagarwaL-code** [comment]
Test it our for every edge case you can think of and write it out here in this form 


- User profile with login
- User profile without login
- ability to share profile


etc

**Comment by devsharmagit** [comment]
@VineeTagarwaL-code 
okay
I will test it heavily and share the results with you.

**Comment by VineeTagarwaL-code** [comment]
no results yet ?

**Comment by devsharmagit** [comment]
@VineeTagarwaL-code 
can we connect over discord ? since there are lot of moving pieces. I would like to clarify them first.

**Comment by VineeTagarwaL-code** [comment]
yeah yeah why not

**Comment by VineeTagarwaL-code** [comment]
build failed

**Comment by devsharmagit** [comment]
@VineeTagarwaL-code  
fixed

**Comment by VineeTagarwaL-code** [comment]
give a video ref demonstrating the whole flow

**Comment by VineeTagarwaL-code** [comment]
and fix the conflicts too

**Comment by devsharmagit** [comment]
https://drive.google.com/file/d/1lX7gi_881D86Yk3wl52jBepf2ttHJIWG/view?usp=sharing

here is the video

**Comment by VineeTagarwaL-code** [comment]
we can improve the code quality by getting rid of repititions of delete dialogs and stuff, but do it in a other pr merging this for now 

great work !

**Comment by devsharmagit** [comment]
@VineeTagarwaL-code 
thanks for merging
sure I will refactor the code.
